### PR TITLE
indexIndicatorFormatFn option

### PR DESCRIPTION
**Pull request by fabiosantoscode**
I needed this for improving promos.sapo.pt (the new version uses photoswipe, but it's not out yet!), so I decided to contribute it back!

What do you think?

Thanks for an amazing, performant, compatible and robust gallery!

### CODE REVIEW

**Comment by dimsemenov** [comment]
Thank you, but it's possible to override public functions of UI directly –  https://github.com/dimsemenov/PhotoSwipe/issues/819#issuecomment-94652574 

So I'm not sure if your modification is needed.

**Comment by fabiosantoscode** [comment]
It does appear like that is the case, but I never saw it documented anywhere, which drove me to send this PR. Maybe overwriting these functions could be a documented feature?

OTOH this amounts to monkey patching, unless the interface is somewhat stabilized.

File: src/js/ui/photoswipe-ui-default.js [hunk 1]
```diff
@@ -707,9 +707,16 @@ var PhotoSwipeUI_Default =
...
 		if(_options.counterEl) {
-			_indexIndicator.innerHTML = (pswp.getCurrentIndex()+1) + 
-										_options.indexIndicatorSep + 
-										_options.getNumItemsFn();
+			var current = pswp.getCurrentIndex(),
+				total = _options.getNumItemsFn(),
+				indicatorHTML;
+
+			if (_options.indexIndicatorFormatFn) {
+				indicatorHTML = _options.indexIndicatorFormatFn(current, total);
```
Comment (muescha) [inline src/js/ui/photoswipe-ui-default.js]: a) `indexIndicatorSep` should be also included as parameter
b) `current` should be with `+1`

**Comment by muescha** [comment]
i like this PR - it is less a monkey patching

File: src/js/ui/photoswipe-ui-default.js [hunk 2]
```diff
@@ -707,9 +707,16 @@ var PhotoSwipeUI_Default =
...
+			} else {
+				indicatorHTML = (current+1) + _options.indexIndicatorSep + total;
+			}
+			_indexIndicator.innerHTML = indicatorHTML;
 		}
 	};
```
Comment (muescha) [inline src/js/ui/photoswipe-ui-default.js]: i would suggest this:

```ruby
_defaultUIOptions = {
    indexIndicatorSep: ' / ',
    indexIndicatorFormatFn: function(current, sep, total){
        return current + sep + total;
    }
}
```

```ruby
	ui.updateIndexIndicator = function() {
		if(_options.counterEl) {
			_indexIndicator.innerHTML =  _options.indexIndicatorFormatFn(pswp.getCurrentIndex()+1,
											 _options.indexIndicatorSep, 
											 _options.getNumItemsFn());
		}
	};
```

**Pull request closed without merging**